include/pybind11/cast.h

```diff
@@ -837,6 +864,57 @@ template <typename T> object cast(const T &value,
 template <typename T> T handle::cast() const { return pybind11::cast<T>(*this); }
 template <> inline void handle::cast() const { return; }
 
+template <typename T>
+typename std::enable_if<detail::move_always<T>::value || detail::move_if_unreferenced<T>::value, T>::type move(object &&obj) {
```

Comment:
I threw them on there to protect external callers from accessing it, because `move()` isn't in `detail`, i.e. a pybind11-caller can explicitly ask for the instance to be moved by calling `py::move<T>(object)` as a shortcut for `py::cast<T>(std::move(object))`.